core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -127,6 +127,10 @@ class MetadataCache(brokerId: Int) extends Logging {
     }
   }
 
+  private def hasAliveEndpoint(snapshot: MetadataSnapshot, brokerId: Int, listenerName: ListenerName): Boolean = {
+    snapshot.aliveNodes.get(brokerId).exists(_.contains(listenerName))
```

Comment:
May be worth mentioning that we added this method for perf reasons and it has the same semantics as `getAliveEndpoint` (which has a comment explaining how it works). Otherwise, I can see someone removing this method in a refactoring.